Bottom: 24c795e5bad7d7c53b2ef21274f37d815ce26584
Top:    1eefe3cda03df6d4561ff24000932c3ea0d4e8f3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-09 12:12:24 +0100

Debug faulty outline generation


---

diff --git a/src/polygon.c b/src/polygon.c
index d379a6b..0040ec2 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1959,6 +1959,7 @@ PolyToPolygonsOnLayer (DataType *Destination, LayerType *Layer,
 
 struct clip_outline_info {
   POLYAREA *poly;
+  int line_count;
 };
 
 #define ROUTER_THICKNESS 1000
@@ -1987,9 +1988,13 @@ line_outline_callback (const BoxType * b, void *cl)
   struct clip_outline_info *info = cl;
   POLYAREA *np, *res;
 
+  info->line_count++;
+
   if (!(np = LinePoly (line, ROUTER_THICKNESS)))
     return 0;
 
+  printf ("Subtracting a line\n");
+
   poly_Boolean_free (info->poly, np, &res, PBO_SUB);
   info->poly = res;
 
@@ -2082,9 +2087,11 @@ POLYAREA *board_outline_poly ()
   region.Y1 = 0;
   region.X2 = PCB->MaxWidth;
   region.Y2 = PCB->MaxHeight;
-
+printf ("start\n");
+  info.line_count = 0;
   r_search (Layer->line_tree, &region, NULL, line_outline_callback, &info);
   r_search (Layer->arc_tree,  &region, NULL, arc_outline_callback, &info);
+printf ("end\n");
 
   clipped = info.poly;
 
@@ -2099,6 +2106,7 @@ POLYAREA *board_outline_poly ()
    *          pieces, hopefully the outline layer isn't too complex!
    */
 
+#if 0
   piece = clipped;
   do { /* LOOP OVER POLYGON PIECES */
 
@@ -2121,6 +2129,7 @@ POLYAREA *board_outline_poly ()
   } while ((piece = piece->f) != clipped);
 
   g_list_foreach (pieces_to_delete, delete_piece_cb, &clipped);
+#endif
 
   return clipped;
 }
